x/config/shadowsocks.go

```diff
@@ -144,11 +144,14 @@ func parseShadowsocksSIP002URL(url *url.URL) (*shadowsocksConfig, error) {
 		return nil, errors.New("host not specified")
 	}
 	config.serverAddress = url.Host
-	cipherInfoBytes, err := base64.URLEncoding.WithPadding(base64.NoPadding).DecodeString(url.User.String())
-	if err != nil {
-		return nil, fmt.Errorf("failed to decode cipher info [%v]: %w", url.User.String(), err)
-	}
-	cipherName, secret, found := strings.Cut(string(cipherInfoBytes), ":")
+	cipherInfo := url.User.String()
+	// Cipher info can be optionally encoded with Base64URL.
+	encoding := base64.URLEncoding.WithPadding(base64.NoPadding)
+	decodedCipherInfo, err := encoding.DecodeString(cipherInfo)
+	if err == nil && encoding.EncodeToString(decodedCipherInfo) == cipherInfo {
+		cipherInfo = string(decodedCipherInfo)
+	}
```

Comment:
Nit: I find it clearer to understand when we only assign to cipherInfo when we know it's a cipherInfo, so the semantics of the variable stays constant. Now your initial variable means userInfo, not cipherInfo.
```suggestion
        var cipherInfo string
	if err == nil {
		cipherInfo = string(decodedUserInfo)
	} else {
	 	cipherInfo = userInfo	  
	}
```